spec/system/publishers/publishers_can_manage_self_disclosure_spec.rb

```diff
@@ -159,26 +160,41 @@
        .and not_change(SelfDisclosure, :count)
     end
 
-    describe "visit jobseeker self disclosure form" do
-      let(:request) { create(:self_disclosure_request, job_application_id: job_application.id) }
-
-      before { request.manual! }
+    describe "visit jobseeker self disclosure form", :versioning do
+      before do
+        create(:self_disclosure_request, :manual, job_application_id: job_application.id)
+        publisher_ats_self_disclosure_page.load(
+          vacancy_id: vacancy.id,
+          job_application_id: job_application.id,
+        )
+      end
 
       context "when request is pending" do
+        it "handles errors" do
+          publisher_ats_self_disclosure_page.goto_references_and_self_disclosure_form.click
+          click_on "Save and continue"
+          expect(publisher_ats_self_disclosure_page.errors.map(&:text))
+            .to eq(["Select yes if you would like to collect self disclosure through the service"])
+        end
+
+        it "copes with saying no" do
+          publisher_ats_self_disclosure_page.goto_references_and_self_disclosure_form.click
+          choose "No"
+          click_on "Save and continue"
+          expect(publisher_ats_pre_interview_checks_page).to be_displayed
+        end
+
         it "publisher can go to the references and self_disclosure form" do
-          publisher_ats_self_disclosure_page.load(
-            vacancy_id: vacancy.id,
-            job_application_id: job_application.id,
-          )
           expect(publisher_ats_self_disclosure_page.status.text).to eq("pending")
           expect(publisher_ats_self_disclosure_page.button.text).to eq("Manually mark as complete")
           expect(
             publisher_ats_self_disclosure_page.goto_references_and_self_disclosure_form.text,
           ).to eq("Would you like to collect this self-disclosure form through Teaching Vacancies?")
 
           publisher_ats_self_disclosure_page.goto_references_and_self_disclosure_form.click
-
-          expect(page).to have_content("Would you like to collect references and self-disclosure through the Teaching Vacancies service?")
+          choose "Yes"
+          click_on "Save and continue"
+          expect(publisher_ats_pre_interview_checks_page).to be_displayed
```

Comment:
It feels like there is something else we can assert other than the page the user lands on in this spec. At the moment the final assertion is the same when the user clicks "Yes" as when they select "No". I imagine there is something else we could assert to ensure that the value has been saved.